From b73f4e82d8381a2a8cb6fa01619789abae6b8ea1 Mon Sep 17 00:00:00 2001
From: Jan Grulich <jgrulich@redhat.com>
Date: Mon, 28 May 2018 10:58:27 +0200
Subject: Make sure we open native dialogs when opening folders inside sandbox

We have to take obsoleted QFileDialog::DirectoryOnly file mode option
into account, because it is actually used, even internally in Qt when
using static QFileDialog::getExistingDirectory() method.

Change-Id: I6cdd35ae4724a1d889a0fb1c3555dc3ca4f3bf4d

diff --git a/src/plugins/platformthemes/flatpak/qflatpakfiledialog.cpp b/src/plugins/platformthemes/flatpak/qflatpakfiledialog.cpp
index ce835eaa23..b170abbdf6 100644
--- a/src/plugins/platformthemes/flatpak/qflatpakfiledialog.cpp
+++ b/src/plugins/platformthemes/flatpak/qflatpakfiledialog.cpp
@@ -289,7 +289,7 @@ QUrl QFlatpakFileDialog::directory() const
 {
     Q_D(const QFlatpakFileDialog);

-    if (d->nativeFileDialog && options()->fileMode() == QFileDialogOptions::Directory)
+    if (d->nativeFileDialog && (options()->fileMode() == QFileDialogOptions::Directory || options()->fileMode() == QFileDialogOptions::DirectoryOnly))
         return d->nativeFileDialog->directory();

     return d->directory;
@@ -309,7 +309,7 @@ QList<QUrl> QFlatpakFileDialog::selectedFiles() const
 {
     Q_D(const QFlatpakFileDialog);

-    if (d->nativeFileDialog && options()->fileMode() == QFileDialogOptions::Directory)
+    if (d->nativeFileDialog && (options()->fileMode() == QFileDialogOptions::Directory || options()->fileMode() == QFileDialogOptions::DirectoryOnly))
         return d->nativeFileDialog->selectedFiles();

     QList<QUrl> files;
@@ -345,7 +347,7 @@ void QFlatpakFileDialog::exec()
 {
     Q_D(QFlatpakFileDialog);

-    if (d->nativeFileDialog && options()->fileMode() == QFileDialogOptions::Directory) {
+    if (d->nativeFileDialog && (options()->fileMode() == QFileDialogOptions::Directory || options()->fileMode() == QFileDialogOptions::DirectoryOnly)) {
         d->nativeFileDialog->exec();
         return;
     }
@@ -374,7 +376,7 @@ bool QFlatpakFileDialog::show(Qt::WindowFlags windowFlags, Qt::WindowModality wi
     d->modal = windowModality != Qt::NonModal;
     d->winId = parent ? parent->winId() : 0;

-    if (d->nativeFileDialog && options()->fileMode() == QFileDialogOptions::Directory)
+    if (d->nativeFileDialog && (options()->fileMode() == QFileDialogOptions::Directory || options()->fileMode() == QFileDialogOptions::DirectoryOnly))
         return d->nativeFileDialog->show(windowFlags, windowModality, parent);

     openPortal();
